packages/p2p/src/utils/sendbird-api.js

```diff
@@ -0,0 +1,93 @@
+import SendBird from 'sendbird';
+
+class SendbirdAPI {
+    constructor({ LocalStorage }) {
+        this.LocalStorage = LocalStorage;
+        this.channels = {};
+        this.sendbird = null;
+    }
+
+    async init(sendbird_user_id, service_token) {
+        this.sendbird = new SendBird({ appId: '4E259BA5-C383-4624-89A6-8365E06D9D39' });
+        this.sendbird.connect(sendbird_user_id, service_token, (user, error) => {
+            if (error) {
+                // eslint-disable-next-line no-console
+                console.warn({ error });
+                return;
+            }
+
+            const channel_event_handler = new this.sendbird.ChannelHandler();
+            const user_event_handler = new this.sendbird.UserEventHandler();
+
+            channel_event_handler.onMessageReceived = this.onMessageReceived.bind(this);
+            channel_event_handler.onChannelChanged = this.onMessageReceived.bind(this);
+
+            this.sendbird.addChannelHandler('channel_event_handler', channel_event_handler);
+            this.sendbird.addUserEventHandler('user_event_handler', user_event_handler);
+            this.syncChannels();
+        });
+
+        return Promise.resolve();
+    }
+
+    async disconnect() {
+        if (this.sendbird) {
+            this.sendbird.disconnect();
+        }
+
+        return Promise.resolve();
+    }
+
+    async syncChannels() {
+        const channels = {};
+        const channel_list_query = this.sendbird.GroupChannel.createMyGroupChannelListQuery();
+
+        channel_list_query.includeEmpty = true;
+        channel_list_query.limit = 100;
+
+        while (channel_list_query.hasNext) {
+            if (channel_list_query.isLoading) {
+                /* eslint-disable */
+                await new Promise(resolve => setTimeout(resolve, 100));
+                continue;
+                /* eslint-enable */
+            }
+
+            // eslint-disable-next-line no-loop-func
+            channel_list_query.next((channel_list, error) => {
+                if (!error) {
+                    channel_list.forEach(channel => {
+                        this.LocalStorage.setNotificationByChannelUrl(channel.url, {
+                            unread_msgs: channel.unreadMessageCount,
+                        });
+                    });
+                }
+            });
+
+            if (!channel_list_query.hasNext) {
+                break;
+            }
```

Comment:
`while (channel_list_query.hasNext) {` isn't this line of code indicate that when there's no next item, the loop will break?

Reply:
I'm not sure what you mean here. You're right, the loop will `break` when there is no next item. `channel_list_query.next` will move the iterator so the value changes between line 48 and 67.

Reply:
Line 70-72 is not required. It will break the loop anyway during the next while condition check.